### PR TITLE
MTD Validation: update track validation, adding TDR-like study of extrapolation at BTL surface

### PR DESCRIPTION
**Pull request by fabiocos**
#### PR description:

This PR proposes a reorganization of the ```MtdTracksValidation``` class in ```Validation/MtdValidation``` aiming at:

- simplify the code for matching ```TrackingParticle``` and ```PSimHit``` from MTD;
- add a dedicated test to extrapolation of tracks to the BTL surface, parallel to the studies proposed in the TDR, for direct comparison.

This update serves also as a basis for a private backport to the original TDR release 10_4_0_mtd5, for direct comparison with TDR results.

#### PR validation:

The code has been extensively tested on large (100k events) samples of muon and pion guns.

### CODE REVIEW

**Comment by fabiocos** [comment]
please test

**Comment by fabiocos** [comment]
The changes are in the plots that have been affected by this PR

@ctarricone FYI

**Comment by emanueleusai** [comment]
+1

- changes as expected

File: Validation/MtdValidation/plugins/MtdTracksValidation.cc [hunk 1]
```diff
@@ -33,6 +34,31 @@
 #include "Geometry/MTDGeometryBuilder/interface/ProxyMTDTopology.h"
 #include "Geometry/MTDGeometryBuilder/interface/RectangularMTDTopology.h"
 
+#include "RecoMTD/DetLayers/interface/MTDDetLayerGeometry.h"
+#include "RecoMTD/Records/interface/MTDRecoGeometryRecord.h"
```
Comment (perrotta) [inline Validation/MtdValidation/plugins/MtdTracksValidation.cc:38]:
```suggestion
```
This is duplicated below

**Comment by perrotta** [comment]
+1

**Pull request merged**